src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpRequestStream.cs

```diff
@@ -218,13 +224,8 @@ private Task InternalWriteAsync(byte[] buffer, int offset, int count, Cancellati
 
         private Task<bool> InternalWriteDataAsync(byte[] buffer, int offset, int count, CancellationToken token)
         {
-            Debug.Assert(count >= 0);
-            
-            if (count == 0)
-            {
-                return Task.FromResult<bool>(true);
-            }
```

Comment:
InternalWriteDataAsync is also called from EndUploadAsync... any value in this fast path still being kept for that?  If so, we may also want to cache this in a readonly static rather than allocating it each time.